GameServerLib/GameObjects/Spell/Spell.cs

```diff
@@ -368,6 +363,28 @@ public bool Cast(Vector2 start, Vector2 end, IAttackableUnit unit = null)
 
             Script.OnSpellPreCast(CastInfo.Owner, this, unit, start, end);
 
+            if (_game.Config.GameFeatures.HasFlag(FeatureFlags.EnableManaCosts))
+            {
+                // if the spell is targeted
+                if (targetingType == TargetingType.Target)
+                {
+                    var distance = Vector2.DistanceSquared(CastInfo.Owner.Position, unit.Position);
+                    var castRange = GetCurrentCastRange();
+                    // and the targeted unit is in range
+                    if (distance <= castRange * castRange)
```

Comment:
This shouldn't be necessary as it is maintained by previous checks. If it is still necessary, fix the previous checks rather than adding this one.